tfjs-converter/src/executor/graph_executor.ts

```diff
@@ -380,19 +382,23 @@ export class GraphExecutor implements FunctionExecutor {
   private checkTensorForDisposalWithNodeLiveUntilInfo(
       node: Node, tensorMap: NamedTensorsMap, context: ExecutionContext,
       tensorsToKeep: Set<number>, outputNodeNameSet: Set<string>,
-      liveUntilNodes?: string[]) {
-    // Skip output nodes and any control flow nodes, since its dependency is
-    // tricky to track correctly.
-    if (isControlFlow(node) || outputNodeNameSet.has(node.name)) {
-      return;
+      liveUntilNodes?: Node[]) {
+    function isNonDisposableNode(node: Node) {
+      // Skip output nodes and any control flow nodes, since its dependency is
+      // tricky to track correctly.
+      return isControlFlow(node) || outputNodeNameSet.has(node.name);
     }
-    if (liveUntilNodes == null) {
+
+    if (isControlFlow(node) || liveUntilNodes == null) {
       return;
     }
 
-    for (const inputNodeName of liveUntilNodes) {
-      const tensors =
-          getTensorsForCurrentContext(inputNodeName, tensorMap, context);
+    for (const liveUntilThisNode of liveUntilNodes) {
+      if (isNonDisposableNode(liveUntilThisNode)) {
+        continue;
+      }
+      const tensors = getTensorsForCurrentContext(
+          liveUntilThisNode.name, tensorMap, context);
```

Comment:
Nit on the naming: For a node `n_k` in the graph, `liveUntilNodes_k` is the set of nodes `{n_a, n_b, ...}` that will be disposed immediately after `n_k` is run. 

```suggestion
    for (const nodeToDispose of liveUntilNodes) {
      if (isNonDisposableNode(nodeToDispose)) {
        continue;
      }
      const tensors = getTensorsForCurrentContext(
          nodeToDispose.name, tensorMap, context);
```